tests/integration/lineage.js

```diff
@@ -0,0 +1,489 @@
+      expect(db.query.callCount).to.equal(1);
+      expect(db.query.getCall(0).args[0]).to.equal('medic-client/docs_by_id_lineage');
+      expect(db.query.getCall(0).args[1].startkey).to.deep.equal([ id ]);
+      expect(db.query.getCall(0).args[1].endkey).to.deep.equal([ id, {} ]);
+      expect(db.query.getCall(0).args[1].include_docs).to.deep.equal(true);
+      expect(result).to.have.lengthOf(2);
+      expect(result[0]).excluding('_rev').to.deep.equal(one_parent);
+      expect(result[1]).excluding('_rev').to.deep.equal(dummyDoc);
+    });
+  });
+
+  it('fetchContacts fetches contacts with correct parameters', function() {
+    sinon.spy(db, 'allDocs');
+    const fakeLineage = [
+      { _id: 'abc', contact: { _id: 'def' }, parent: { _id: 'ghi' } },
+      { _id: 'ghi' }
+    ];
+    return lineage.fetchContacts(fakeLineage).then(() => {
+      expect(db.allDocs.callCount).to.equal(1);
+      expect(db.allDocs.getCall(0).args[0]).to.deep.equal({ keys: ['def'], include_docs: true });
+    });
+  });
+
+  it('fetchHydratedDoc returns errors from query', function() {
+    return lineage.fetchHydratedDoc('abc').then(
+      () => {
+        throw new Error('should have thrown a 404');
+      },
+      err => {
+        expect(err.status).to.equal(404);
+      }
+    );
+  });
+
+  it('fetchHydratedDoc returns unmodified doc when there is no lineage and no contact', function() {
+    return lineage.fetchHydratedDoc('no_lineageContact').then(actual => {
+      expect(actual).excluding('_rev').to.deep.equal(no_lineageContact);
+    });
+  });
+
+  it('fetchHydratedDoc handles doc with unknown parent by leaving just the stub', function() {
+    return lineage.fetchHydratedDoc(stub_parents._id).then(actual => {
+      expect(actual).excludingEvery('_rev').to.deep.equal({
+        _id: stub_parents._id,
+        type: stub_parents.type,
+        contact: { _id: 'something' },
+        parent: {
+          _id: 'something_else',
+          parent: dummyDoc
+        }
+      });
+    });
+  });
+
+  it('fetchHydratedDoc handles doc with unknown contact by leaving just the stub', function() {
+    return lineage.fetchHydratedDoc(stub_contacts._id).then(actual => {
+      expect(actual).excludingEvery('_rev').to.deep.equal({
+        _id: stub_contacts._id,
+        type: stub_contacts.type,
+        contact: {
+          _id: 'something',
+          parent: dummyDoc
+        },
+        form: {}
+      });
+    });
+  });
+
+  it('fetchHydratedDoc handles missing contacts', function() {
+    return lineage.fetchHydratedDoc(no_contact._id).then(actual => {
+      expect(actual).excluding('_rev').to.deep.equal(no_contact);
+    });
+  });
+
+  it('fetchHydratedDoc attaches the full lineage for reports', function() {
+    return lineage.fetchHydratedDoc(report._id).then(actual => {
+      expect(actual.form).to.equal('A');
+      // parents
+      expect(actual.contact.name).to.equal(report_contact.name);
```

Comment:
Lots of `expect()`s!  Would it be neater if this were a single `expect()` call with a big object as the argument for `.to.equal()`?

Reply:
Yup this was just copied over from the previous unit tests, I agree I'll try and make it into one big `expect`